lib/pacemaker/pcmk_rule.c

```diff
@@ -152,11 +152,13 @@ do_check_rule(pcmk__output_t *out, pe_working_set_t *data_set,
     free(xpath);
     freeXpathObject(xpath_obj);
 
-    if (num_results == 0) {
-        out->err(out, "No rule found with ID=%s", rule_id);
-        return ENXIO;
-    } else if (num_results > 1) {
-        out->err(out, "More than one rule with ID=%s found", rule_id);
+    if (num_results != 1) {
+        if (num_results == 0) {
+            out->err(out, "No rule found with ID=%s", rule_id);
+        } else {    // num_results > 1
+            out->err(out, "More than one rule with ID=%s found", rule_id);
+        }
+        out->message(out, "rule-check", rule_id, ENXIO);
         return ENXIO;
```

Comment:
We have pcmk_rc_multiple but unfortunately that's specifically for "Resource active on multiple nodes". I think this sort of thing comes up enough to be worth a more generic code like pcmk_rc_duplicate_id.

Reply:
agreed, done